Sources/Utility/ColorWrap.swift

```diff
@@ -0,0 +1,88 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright 2015 - 2016 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+
+import func POSIX.isatty
+import libc
+
+public var colorMode = ColorWrap.Mode.Auto 
+
+/// Wrap the text with color.
+public enum ColorWrap {
+
+    /// Wrap the given text with provided color for a stream.
+    /// Color codes will only be added if and only if:
+    ///     stream is TTY && --color=auto
+    ///     or
+    ///     --color=always
+    public static func wrap(_ input: String, with color: Color, for stream: Stream) -> String {
+        guard ColorWrap.isAllowed(for: .StdErr) else { return input }
```

Comment:
@bhargavg @ddunbar Should the `for: .StdErr` here be the currently unused `stream` argument?